shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/cart/AddCartProductUseCase.kt

```diff
@@ -1,24 +1,60 @@
 package com.bunbeauty.shared.domain.feature.cart
 
 import com.bunbeauty.shared.Constants.CART_PRODUCT_LIMIT
+import com.bunbeauty.shared.data.repository.AdditionGroupRepository
+import com.bunbeauty.shared.data.repository.AdditionRepository
+import com.bunbeauty.shared.data.repository.CartProductAdditionRepository
+import com.bunbeauty.shared.domain.feature.addition.GetAdditionPriorityUseCase
+import com.bunbeauty.shared.domain.feature.addition.GetIsAdditionsAreEqualUseCase
+import com.bunbeauty.shared.domain.model.addition.Addition
+import com.bunbeauty.shared.domain.model.addition.AdditionGroup
+import com.bunbeauty.shared.domain.model.cart.CartProduct
 import com.bunbeauty.shared.domain.repo.CartProductRepo
 
 class AddCartProductUseCase(
     private val cartProductRepo: CartProductRepo,
+    private val cartProductAdditionRepository: CartProductAdditionRepository,
+    private val additionRepository: AdditionRepository,
+    private val getIsAdditionsAreEqualUseCase: GetIsAdditionsAreEqualUseCase,
+    private val additionGroupRepository: AdditionGroupRepository,
+    private val getAdditionPriorityUseCase: GetAdditionPriorityUseCase,
 ) {
-
-    suspend operator fun invoke(menuProductUuid: String): Boolean {
+    suspend operator fun invoke(
```

Comment:
We use this usecase 3 times: on Menu, on Cart, on ProductDetails. But actually we should use it only on ProductDetails. 
For Menu we should use simpler logic without checking `Additions`. 
For Cart we shoud add product by `CartProductUuid`

Reply:
✅ fixed